src/libsyntax/mut_visit.rs

```diff
@@ -309,7 +310,12 @@ pub trait MutVisitor: Sized {
 //
 // No `noop_` prefix because there isn't a corresponding method in `MutVisitor`.
 pub fn visit_clobber<T, F>(t: &mut T, f: F) where F: FnOnce(T) -> T {
-    unsafe { std::ptr::write(t, f(std::ptr::read(t))); }
+    unsafe {
```

Comment:
Could you leave a note as to why it is safe?
(we should really make tidy require this at some point... it's troubling that this information is lacking in the compiler of all places...)

Reply:
I added a note. I wasn't sure how to explain it, suggestions are welcome.